src/complex.cr

```diff
@@ -130,14 +175,10 @@ struct Complex
            @imag.abs / Math.sqrt(2 * (r + @real))
          end
 
-    if @imag >= 0
-      Complex.new(re, im)
-    else
-      Complex.new(re, -im)
-    end
+    Complex.new(re, @imag >= 0 ? im : -im)
```

Comment:
`im.abs`?

Reply:
```suggestion
    Complex.new(re, im.abs)
```

Reply:
```suggestion
    Complex.new(re, im.abs)
```

Reply:
Condition here is based on `@imag`, not `im`, so it wouldn't work (if I grok your suggestion correctly).

Reply:
My bad, I misread.